### PR TITLE
Single particle with multiple layers 

### PR DESCRIPTION
**Pull request by jtgasparik**

### CODE REVIEW

File: src/aero_reps/aero_rep_single_particle.F90 [hunk 1]
```diff
@@ -847,8 +847,8 @@ function ordered_layer_ids(layer_names_unordered, cover_names_unordered)
     do i_cover = 2, size(ordered_layer_ids)
       do i_layer = 1, size(layer_names_unordered)
         if (layer_names_unordered(ordered_layer_ids(i_cover-1))%string &
-            .eq. cover_names_unordered(i_layer)%string) then
-          ordered_layer_ids(i_cover) = i_layer
+            .eq. cover_names_unordered(i_layer)%string) then
```
Comment (jcurtis2) [inline src/aero_reps/aero_rep_single_particle.F90]:
```suggestion
            .eq. cover_names_unordered(i_layer)%string) then
```

File: test/unit_aero_rep_data/test_aero_rep_single_particle.F90 [hunk 2]
```diff
@@ -193,12 +195,12 @@ subroutine test_config_read()
         ! test num_phase_instances funtion
         phase_name_test = "bread"
         num_bread = 2
-        max_part = aero_rep%maximum_computational_particles()
+        max_part = aero_rep%maximum_computational_particles()
```
Comment (jcurtis2) [inline test/unit_aero_rep_data/test_aero_rep_single_particle.F90]:
```suggestion
        max_part = aero_rep%maximum_computational_particles()
```

File: test/unit_aero_rep_data/test_aero_rep_single_particle.F90 [hunk 3]
```diff
@@ -412,16 +417,15 @@ logical function build_aero_rep_data_set_test()
 
     deallocate(buffer)
     deallocate(aero_rep_passed_data_set)
-#endif
 
+#endif
     ! Evaluate the aerosol representation c functions
     build_aero_rep_data_set_test = eval_c_func(camp_core)
-
+    print *, 'building'
```
Comment (jcurtis2) [inline test/unit_aero_rep_data/test_aero_rep_single_particle.F90]: Remove

File: src/aero_reps/aero_rep_single_particle.c [hunk 4]
```diff
@@ -424,23 +431,24 @@ void aero_rep_single_particle_print(int *aero_rep_int_data,
...
-    printf("\n  state id: %d model data id: %d num Jac elements: %d",
-           PHASE_STATE_ID_(i_phase), PHASE_MODEL_DATA_ID_(i_phase),
-           PHASE_NUM_JAC_ELEM_(i_phase));
-  }
+  for(int i_layer = 0; i_layer < NUM_LAYERS_; ++i_layer){
+    for (int i_phase = 0; i_phase < NUM_PHASES_(i_layer); ++i_phase) {
+      printf("\n  state id: %d model data id: %d num Jac elements: %d",
+             PHASE_STATE_ID_(i_layer,i_phase), PHASE_MODEL_DATA_ID_(i_layer,i_phase),
+             PHASE_NUM_JAC_ELEM_(i_layer,i_phase));
+    }
```
Comment (mattldawson) [inline src/aero_reps/aero_rep_single_particle.c:446]: Do you want to modify this to output something like?:

```
  - Layers -
  layer 1
    - Phases -
      state id:...
      state id:...
  layer 2
    - Phases -
      state id:...
```

File: test/unit_aero_rep_data/test_aero_rep_single_particle.F90 (hunk 3 above)
Comment (mattldawson) [inline test/unit_aero_rep_data/test_aero_rep_single_particle.F90]: is this still needed?

**Review by mattldawson — APPROVED** [review]
looks great! Just a couple minor questions, but nothing that would prevent merging this in

**Review by jcurtis2 — APPROVED** [review]

File: src/aero_reps/aero_rep_single_particle.F90 (hunk 1 above)
Comment (jtgasparik) [reply src/aero_reps/aero_rep_single_particle.F90:850]: what changed here?

File: src/aero_reps/aero_rep_single_particle.c (hunk 4 above)
Comment (jtgasparik) [reply src/aero_reps/aero_rep_single_particle.c:446]: can you clarify what you mean by this?

**Pull request merged**